src/server/routers/tips.ts

```diff
@@ -44,16 +45,17 @@ export const tipsRouter = router({
             '1234567890abcdefghijklmnopqrstuvwxyz',
             5,
           )
-
-          const {instructor_id: instructorId} = JSON.parse(
-            ctx.req?.cookies.get('eh_user')?.value ?? "{instructor_id: ''}",
+          const {instructor_id: instructorId} = await loadCurrentUser(
+            ctx?.userToken,
           )
+          if (!instructorId) throw new Error('Instructor ID not found')
 
           let sanityInstructor = await sanityWriteClient.fetch(groq`
           *[_type == 'collaborator' && eggheadInstructorId == '${instructorId}'][0] {
             _id,
-          }
-        `)
+          }`)
+          if (!sanityInstructor)
+            throw new Error('Sanity Instructor ID not found')
```

Comment:
sanity instructor could also possibly not be set so this returns an error if that's the case